spec/govuk_error/configuration_spec.rb

```diff
@@ -12,36 +12,69 @@
   end
 
   describe ".should_capture" do
-    around do |example|
-      ClimateControl.modify GOVUK_DATA_SYNC_PERIOD: "22:00-08:00" do
-        example.run
-      end
-    end
-
     let(:configuration) { GovukError::Configuration.new(Raven.configuration) }
 
-    it "should capture errors that occur during the data sync" do
-      travel_to(Time.current.change(hour: 23)) do
+    context "during the data sync" do
+      around do |example|
+        ClimateControl.modify GOVUK_DATA_SYNC_PERIOD: "22:00-08:00" do
+          travel_to(Time.current.change(hour: 23)) do
+            example.run
+          end
+        end
+      end
+
+      it "should capture errors by default" do
         expect(configuration.should_capture.call(StandardError.new)).to eq(true)
       end
-    end
 
-    it "should ignore errors that have been added to data_sync_excluded_exceptions, if they occurred during the data sync" do
-      configuration.data_sync_excluded_exceptions << "StandardError"
+      it "should ignore errors that have been added to data_sync_excluded_exceptions" do
+        configuration.data_sync_excluded_exceptions << "StandardError"
 
-      travel_to(Time.current.change(hour: 23)) do
         expect(configuration.should_capture.call(StandardError.new)).to eq(false)
       end
+
+      it "should ignore errors whose underlying cause is an exception in data_sync_excluded_exceptions" do
+        stub_const("PG::Error", Class.new(StandardError))
+        stub_const("SomeOtherError", Class.new(StandardError))
+        configuration.data_sync_excluded_exceptions << "PG::Error"
+
+        chained_exception = nil
+        begin
+          begin
+            raise PG::Error
+          rescue PG::Error
+            raise SomeOtherError
+          end
+        rescue SomeOtherError => e
+          chained_exception = e
+        end
+
+        expect(chained_exception).to_not be_nil
```

Comment:
I guess this is added because the test would still pass if chained_exception had its initial value. I think if you're going to assert on it you may as well assert that it is the class you're expecting. Eg `expect(chained_exception).to be_an_instance_of(SomeOtherError)`